plugins/ipstack/icon_ipstack/connection/connection.py

```diff
@@ -13,4 +16,19 @@ def connect(self, params):
         self.token = params.get(Input.CRED_TOKEN).get("secretKey")
 
     def test(self):
-        return {"success": True}
+        # Sending a request using the secretKey to test if it is valid
+        url = "http://api.ipstack.com/" + "rapid7.com" + "?access_key=" + self.token + "&output=json"
+
+        # These contain custom status codes (101, 102, 104, 106) returned from IPStack, as seen in the lookup action.py
+        invalid_status_codes = ["404", "101", "102", "104", "106"]
+
+        resp = insightconnect_plugin_runtime.helper.open_url(url)
+        dic = json.loads(resp.read())
+        if any(invalid_status_codes) in dic or "error" in dic:
+            raise PluginException(
+                data=dic,
+                assistance="Please check your credentials and try again. If the issue persists, please contact support.",
+                cause="An error has occurred. Please check your credentials are correct and that you have not exceeded the lookup limits.",
+            )
+        else:
+            return {"success": True}
```

Comment:
Not necessary `else`
```suggestion
        return {"success": True}
```